src/Npgsql/TypeMapping/ConnectorTypeMapper.cs

```diff
@@ -16,30 +17,26 @@ namespace Npgsql.TypeMapping
 {
     sealed class ConnectorTypeMapper : TypeMapperBase
     {
-        /// <summary>
-        /// The connector to which this type mapper belongs.
-        /// </summary>
         readonly NpgsqlConnector _connector;
 
         NpgsqlDatabaseInfo? _databaseInfo;
 
-        /// <summary>
-        /// Type information for the database of this mapper.
-        /// </summary>
         internal NpgsqlDatabaseInfo DatabaseInfo
-            => _databaseInfo ?? throw new InvalidOperationException("Internal error: this type mapper hasn't yet been bound to a database info object");
+        {
+            get => _databaseInfo ?? throw new InvalidOperationException("Internal error: this type mapper hasn't yet been bound to a database info object");
+            set => _databaseInfo = value;
+        }
 
         internal NpgsqlTypeHandler UnrecognizedTypeHandler { get; }
 
-        readonly Dictionary<uint, NpgsqlTypeHandler> _byOID = new();
-        readonly Dictionary<NpgsqlDbType, NpgsqlTypeHandler> _byNpgsqlDbType = new();
-        readonly Dictionary<DbType, NpgsqlTypeHandler> _byDbType = new();
-        readonly Dictionary<string, NpgsqlTypeHandler> _byTypeName = new();
+        internal IDictionary<string, NpgsqlTypeMapping> MappingsByName { get; private set; }
+        internal IDictionary<NpgsqlDbType, NpgsqlTypeMapping> MappingsByNpgsqlDbType { get; private set; }
+        internal IDictionary<Type, NpgsqlTypeMapping> MappingsByClrType { get; private set; }
 
-        /// <summary>
-        /// Maps CLR types to their type handlers.
-        /// </summary>
-        readonly Dictionary<Type, NpgsqlTypeHandler> _byClrType= new();
+        readonly Dictionary<uint, NpgsqlTypeHandler> _handlersByOID = new();
+        readonly Dictionary<NpgsqlDbType, NpgsqlTypeHandler> _handlersByNpgsqlDbType = new();
+        readonly Dictionary<string, NpgsqlTypeHandler> _handlersByTypeName = new();
+        readonly Dictionary<Type, NpgsqlTypeHandler> _handlersByClrType = new();
 
         /// <summary>
         /// Maps CLR types to their array handlers.
```

Comment:
For the constructor below - we can just call `Reset` instead of `ClearBindings` and `ResetMappings`.